airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java

```diff
@@ -195,9 +196,12 @@ public Optional<UUID> getSourceId(final UUID connectionId) {
   @Override
   public Optional<Status> getStatus(final UUID connectionId) {
     try {
-      final StandardSync standardSync = getStandardSync(connectionId);
-      return Optional.ofNullable(standardSync.getStatus());
-    } catch (final JsonValidationException | ConfigNotFoundException | IOException e) {
+      final io.airbyte.api.client.model.generated.ConnectionIdRequestBody requestBody =
+          new io.airbyte.api.client.model.generated.ConnectionIdRequestBody().connectionId(connectionId);
+      final ConnectionRead connectionRead = connectionApi.getConnection(requestBody);
```

Comment:
Consider extracting to a shared method as this code is now duplicated with the method above.